core/domain/classifier_services_test.py

```diff
@@ -779,11 +778,20 @@ def test_reverted_exploration_maintains_classifier_model_mapping(self):
         with self.swap(feconf, 'ENABLE_ML_CLASSIFIERS', True):
             exp_services.update_exploration(
                 feconf.SYSTEM_COMMITTER_ID, self.exp_id, change_list, '')
+
         current_exploration = exp_fetchers.get_exploration_by_id(self.exp_id)
-        (
-            classifier_services
-            .create_classifier_training_job_for_reverted_exploration(
-            current_exploration, old_exploration))
+        # Store the job before reverting the exploration.
+        job = classifier_services.get_classifier_training_job(
+            self.exp_id, current_exploration.version, state_name,
+            algorithm_id)
+        # Revert the exploration.
+        with self.swap(feconf, 'ENABLE_ML_CLASSIFIERS', True):
+            exp_services.revert_exploration(
+                feconf.SYSTEM_COMMITTER_ID,
+                self.exp_id,
+                current_exploration.version,
+                current_exploration.version - 1
+            )
         # Verify if classifier model mapping is maintained using the job ID.
         new_job = classifier_services.get_classifier_training_job(
```

Comment:
Rather than reusing current_exploration (which is obsolete at this point), better to fetch the exploration again, assert that its version is now current_exploration.version + 1, and then refer to the newly-fetched version when doing the comparison.

Reply:
Done.